UsageEnvironment/Log.cpp

```diff
@@ -132,17 +248,27 @@ void _Log::Fatal(const char* filePath, int line, const char * format, ...) {
 }
 
 void _Log::Error(const char* filePath, int line, const char * format, ...) {
-  if( logLevel >= LogLevelError  ) {
+  if( logLevel >= LogLevelError ) {
     va_list argptr; va_start(argptr, format);
     FILE* fileHandle = AcquireFile();
-    vfprintf(fileHandle, FormatLog(filePath, line, LogLevelError, format).c_str(), argptr);
+    char errorLog[MAX_LOG_LEN];
+    vsnprintf(errorLog, MAX_LOG_LEN, FormatLog(filePath, line, LogLevelError, format).c_str(), argptr);
+    fprintf(fileHandle, "%s", errorLog);
     ReleaseFile(fileHandle);
+#ifdef ERROR_MESSAGE_BACKUP_TRIGGER
+    if( strstr(errorLog, ERROR_MESSAGE_BACKUP_TRIGGER) ) {
+      syslog(LOG_ERR, "RTSP issue triggered.");
+      BackupLog();
+    } else {
+      fprintf(stdout, "%s - log DID NOT trigger backup.\n", errorLog);
+    }
+#endif
```

Comment:
As discussed offline, the client (`rtspd`) will be the one to initiate the trigger. (We do not want to be doing string comparisons on every log message.)